apps/browser/qml/pages/components/OverlayItem.qml

```diff
@@ -0,0 +1,255 @@
+/****************************************************************************
+**
+** Copyright (c) 2021 Open Mobile Platform LLC.
+**
+****************************************************************************/
+
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import QtQuick 2.2
+import QtQuick.Window 2.2 as QuickWindow
+import Sailfish.Silica 1.0
+import Sailfish.Silica.private 1.0 as Private
+import Sailfish.Browser 1.0
+import Sailfish.Policy 1.0
+import Sailfish.WebView.Controls 1.0
+import Sailfish.WebEngine 1.0
+import com.jolla.settings.system 1.0
+import "." as Browser
+import "../../shared" as Shared
+
+SilicaFlickable {
```

Comment:
You are approaching the design changes from a wrong angle.

- [ ] Make overlay to open to top (y: 0) -- 1st commit (PR)
- [ ] Add history item below the search field -- 2nd commit (best would be if this would be a separate pull request)
- [ ] Only when user has entered text to the search field use bookmark field model for the historylist - In design meeting earlier we agreed that filtering should be limited to a row (1). This implies that row, repeater, bookmarkfiltermodel, favoriteitem with a contextmenu should do it (we should consider adding max count property limited to the bookmarkfiltermodel to avoid creating too many delegates) -- 3rd commit (again another is good PR). With this you can control easily if the context menu is opened in the header part of the historylist and adjust footer height accordingly.
- [ ] Adjust historylistitem - 4th commit (PR)

As summary, there are at least 4 different changes that you should prepare preferably 4PRs.